src/deploy/functions/args.ts

```diff
@@ -22,15 +34,11 @@ export interface Context {
 
   // Filled in the "prepare" phase.
   config?: projectConfig.ValidatedSingle;
-  functionsSourceV1?: string;
-  functionsSourceV2?: string;
-  runtimeConfigEnabled?: boolean;
   artifactRegistryEnabled?: boolean;
   firebaseConfig?: FirebaseConfig;
 
-  // Filled in the "deploy" phase.
-  sourceUrl?: string;
-  storage?: Record<string, gcfV2.StorageSource>;
+  // Filled in the "prepare" and "deploy" phase.
+  source?: Source;
```

Comment:
This probably needs to be per-codebase, right?

Reply:
Yep. Being done in https://github.com/firebase/firebase-tools/pull/4419.